File: ILT.IHR.DTO/Ticket.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ILT.IHR.DTO
{
    public partial class Ticket : AbstractDataObject
    {
        public int TicketID
        {
            get { return base.RecordID; }
            set { base.RecordID = value; }
        }
        [Required]
        [RegularExpression("^0*[1-9]\\d*$")]
        public int TicketTypeID { get; set; }
        public string TicketType { get; set; }
        public string TicketShort { get; set; }
        [Required]
        [RegularExpression("^0*[1-9]\\d*$")]
        public int RequestedByID { get; set; }
        public string RequestedBy { get; set; }
        //[Required]
        //[RegularExpression("^0*[1-9]\\d*$")]
        public int? ModuleID { get; set; }
        public string ModuleName { get; set; }
        public int? ID { get; set; }
        [Required]
        public string Description { get; set; }
        //[Required]
        //[RegularExpression("^0*[1-9]\\d*$")]
        public int? AssignedToID { get; set; }
        public string AssignedTo { get; set; }
        [Required]
        [RegularExpression("^0*[1-9]\\d*$")]
        public int StatusID { get; set; }
        public string Status { get; set; }
        public DateTime? ResolvedDate { get; set; }
        public string Comment { get; set; }
        [Required]
        public string Title { get; set; }
        public Guid LinkID { get; set; }
    }
}

File: ILT.IHR.APP/ClientApp/src/app/core/interfaces/Ticket.ts
export class ITicket {
    TicketID: number;
    TicketTypeID: number;
    TicketType: string;
    TicketShort: string;
    RequestedByID: number;
    RequestedBy: string;
    ModuleID: number;
    ModuleName: string;
    ID?: number;
    Description: string;
    AssignedToID: string;
    AssignedTo: string;
    StatusID: number;
    Status: string;
    ResolvedDate?: Date;
    Comment: string;
    Title: string;
    LinkID: string;
    CreatedBy: string;
    CreatedDate: Date;
    ModifiedBy: string;
    ModifiedDate: Date;
    TimeStamp: ['']

}
export class ITicketForDisplay {
    ticketID: number;
    ticketTypeID: number;
    ticketType: string;
    ticketShort: string;
    requestedByID: number;
    requestedBy: string;
    moduleID: number;
    moduleName: string;
    iD?: number;
    description: string;
    assignedToID: number;
    assignedTo: string;
    statusID: number;
    status: string;
    resolvedDate?: Date;
    comment: string;
    title: string;
    linkID: string;
    createdBy: string;
    createdDate: Date;
    modifiedBy: string;
    modifiedDate: Date;
    timeStamp: ['']
}
